Cython/Shadow.py

```diff
@@ -461,3 +461,16 @@ def threadid(self):
 import sys
 sys.modules['cython.parallel'] = CythonDotParallel()
 del sys
+
+
+def __getattr__(name, module=__name__):
+    import re
+    match = re.match('^(p+)_([a-zA-Z0-9]+)$', name)
+    if match:
+        depth = len(match.group(1))
+        type = match.group(2)
+
+        if type in int_types + float_types + complex_types + other_types:
```

Comment:
Might it be worth caching this list, or checking each individually? It'll only be used during import yes, but it's better to not be more expensive than necessary.

Reply:
Agreed. Can be done with an additional default argument, for example.
BTW, `type` overrides the builtin name, so I'd rather call the variable `type_name`.